### ISSUE TITLE
Using a {{customVariable}} object throws an Exception when trying to decode it

### ISSUE THREAD

**Issue by limonadev** [peiffer-innovations/json_dynamic_widget#11]
When using a variable `{{custom}}` to encode some object (not bool, number nor string) in some widget, if the builder uses the function `ThemeDecoder.decodeSomething()`, the program throws an exception because the variable is already the object, but the `decodeSomething()` function usually treats internally the `value` argument as a `Map`.

## Steps to reproduce:
1. Register some custom variable in the Registry (I registered a `Rect` variable).
2. Use that variable in any json (I used it in `asset_images.json`)

## JSON example
```json
{
  "type": "scaffold",
  "args": {
    "appBar": {
      "type": "app_bar",
      "args": {
        "title": {
          "type": "text",
          "args": {
            "text": "Asset Images Page"
          }
        }
      }
    },
    "body": {
      "type": "asset_image",
      "args": {
        "centerSlice": "{{customRect}}",
        "name": "##getImageAsset(1)##"
      }
    }
  }
}
```

It's necessary to add the `{{customRect}}` variable to the Registry:
```dart
registry = JsonWidgetRegistry(
  values: {
    'customRect': Rect.largest,
  },
);
```

**Comment by limonadev** [comment]
As a proposal to solve [this](https://github.com/peiffer-innovations/json_dynamic_widget/issues/11#issue-685922969) issue, I've tested this new class to replace each call to `ThemeDecoder.decodeSomething()` where it wants to be used as a {{variable}} or a JSON map structure.

This is the basic class, only with one function (following the "Steps to reproduce", I solved the problem for `Rect` variables but it also can be generalized to all the objects)
```dart
import 'package:flutter/painting.dart';
import 'package:json_theme/json_theme.dart';

class InstanceDecoder {
  static Rect decodeRect(
    dynamic instance, {
    Rect defaultInstance,
    bool validate = true,
  }) {
    Rect result;

    if (instance is Rect) {
      result = instance;
    } else {
      result = ThemeDecoder.decodeRect(
        instance,
        validate: validate,
      );
    }

    result ??= defaultInstance;

**Comment by jpeiffer** [comment]
I'd rather fix this at the source.  So rather than decoding via a net-new class, I'd rather inside of `ThemeDecoder.decodeRect` check if the incoming dynamic object is a `Rect` then.

That way, any framework that uses `json_theme` will be able to pass in an actual instance or a map-like-structure.

**Comment by jpeiffer** [comment]
Fixed in `1.0.4`.  The actual fix was in `json_theme`, and the "fix" here was just updating the dependency to pull in the most recent `json_theme` version + adding this example to the example app itself.
